### ISSUE TITLE
Thoughts on removing 'redis' as a dependency?

### ISSUE THREAD

**Issue by golmansax** [rv-kip/express-redis-cache#130]
Hi, as a follow-up to https://github.com/rv-kip/express-redis-cache/pull/98, what are your thoughts on removing 'redis' as a dependency?

The reason I ask is that this is the only package in our production repo that has 'redis' as a dependency.

Thanks!

**Comment by rawpixel-vincent** [comment]
makes sense, the library shouldn't require as a dependency, it could be added as an optionalDependencies in the package.json along other supported redis library if any, this need to be tested, and need to be worked out for the integration tests, so possibly also needed in the devDependencies.
PR welcomes if possible

**Comment by demsking** [comment]
`redis` should me moved to the `peerDependencies` as it's used as the main dependency by `express-redis-cache`
